### ISSUE TITLE
Github's new PR review system not recognized as updates to PR

### ISSUE THREAD

**Issue by JonathanTech** [ptsochantaris/trailer#295]
Github came out with new features that allow you to create "reviews" that can reject a PR, accept it, or be impartial. This probably is a new API interface or object, but the main result is that v1.5.0 doesn't recognize these events as updates to the PR, thus causing developers to not be notified when their PR's have been accepted through this new mechanism. 

https://help.github.com/articles/about-pull-request-reviews/

**Comment by ptsochantaris** [comment]
Hi @JonathanTech 

If I understand correctly, you'd like to receive a notification when reviews are actioned (accepted or rejected) by reviewers, is that correct?

If so, it totally makes sense, but I don't currently see any API in place to support this (quite possibly it's being created as we speak).

If you think you can spot any API related to this feature, let me know and I'll definitely investigate, as this sounds like a very useful feature to have.

**Comment by JonathanTech** [comment]
Accepted, rejected, or comments on the reviews. It is very likely they don't have a public API for it yet. I'll try to see if there's any documentation and keep this thread updated if I see new ones.

**Comment by ptsochantaris** [comment]
Interesting, I would have thought the comments would have been treated as normal comments in the existing API. Thanks for the info, I'll be keeping an eye out if you find anything.

**Comment by JonathanTech** [comment]
Looks like it's part of the v3 api and should be in the `review_comments` fields.
https://developer.github.com/v3/pulls/

**Comment by JonathanTech** [comment]
Seems like either that field is of the same structure as `comments` or they have yet to write the new structure documentation.

**Comment by ptsochantaris** [comment]
Indeed, `review_comments` is where Trailer currently gets all the comments for PRs currently. The `comments` field is used for comments in issues (i.e. the ones that aren't PRs I mean :))

**Comment by JonathanTech** [comment]
😵  Ok. Looks like I'll have to figure out how to hit the api on my own PR's to see if I can find the delta's for this compared to before.

**Comment by ptsochantaris** [comment]
Cool, I'd definitely be very interested to know if you find out anything more about this.

I'll have a look as well when there's some time, but in the meantime if you *do* spot any of this review mechanism exposed in the current API, give a shout!

**Comment by JonathanTech** [comment]
Ok, yeah, they aren't exposing it yet as far as I can tell. What does show up are comments on code within the review, but not the review summary comments. Also the part of the review that shows up in the statuses list on the html doesn't show up in the api yet either. I'll keep a watch on the api and reopen this when it becomes available.

**Comment by ptsochantaris** [comment]
That would be awesome, looking forward to being able to include this in Trailer when it's there!

**Comment by JonathanTech** [comment]
Email I got after asking github directly about the API and this feature. 
```
Hi Jonathan,

Thanks for getting in touch!

We appreciate your feedback, as well as your interest in such an API for pull request reviews. Adding an API for this is definitely on our radar, and the team shared some thoughts around why an API is not available for some features at the same moment when they're launched:

https://news.ycombinator.com/item?id=12501075

There isn't a timetable that I can share, nor a preview API for which these can be seen since they don't currently exist. I can't promise when support for this will be added, so I'd recommend following the API blog for updates: https://developer.github.com/changes/

If you have any other questions -- please let me know.

All the best,
Francis
```

**Comment by ptsochantaris** [comment]
Thanks for sharing this @JonathanTech, much appreciated. It's pretty much what I imagined the answer would have been, but it's good to have actual confirmation that API support is planned.

It looks like, for the moment, all we can do is be patient. This will definitely be staying on the TODO list of things to add as soon as they're available on the API (right next to reactions support, which still isn't available for production-level use and subject to change).

In the meantime, I'd like to keep this issue open so that (a) Anyone visiting about this issue can find out quicker about its status and (b) It will help keep it as a TODO item on my own Trailer list :)

**Comment by Pedlar** [comment]
@ptsochantaris @JonathanTech GitHub updated on this yesterday.

Blog Post:
https://developer.github.com/changes/2016-10-11-pull-request-review-webhooks/

API Doc:
https://developer.github.com/v3/activity/events/types/#pullrequestreviewevent

**Comment by ptsochantaris** [comment]
Thanks for the update @Pedlar - Great to know that things are moving along on that front.

Trailer does scan the event stream during sync, which includes the same payloads as the web hooks, so I'm wondering if it's at least worth implementing notifications based on the `PullRequestReviewEvent` type.

It still however does not solve the issues with certain comments not showing up as @JonathanTech points out above.

**Comment by akalman** [comment]
looks like this is available now?  review requests as well.

https://developer.github.com/v3/pulls/reviews/

**Comment by ptsochantaris** [comment]
Hi @akalman Indeed, it's been available for preview for a while, but the only thing that would do is help me start implementing something early on - the actual clients can't access this API because it's in flux and hence the API (and the client) could break without warning. This API has been in preview for a long while, hopefully it should settle down sooner rather than later.

**Comment by ptsochantaris** [comment]
Hi, just wanted to update you that there is ongoing work on this feature now. There are beta builds available here for anyone who feels adventurous: #314 

<img width="487" alt="screen shot 2017-03-11 at 12 36 19" src="https://cloud.githubusercontent.com/assets/521581/23823295/7d89fe72-0657-11e7-8462-36a6e5d65526.png">

**Comment by ptsochantaris** [comment]
Hi, I'm going to close this issue as v1.5.6 is up now and includes this feature.

Please feel free to open issues related to Review support, bugs, new behaviour or logic that you'd like to see related to support for reviews if you get a chance to try this feature out.
